pytorch_lightning/utilities/__init__.py

```diff
@@ -16,33 +16,39 @@
 from enum import Enum
 
 import numpy
-import torch
 
 from pytorch_lightning.utilities.apply_func import move_data_to_device
 from pytorch_lightning.utilities.distributed import rank_zero_info, rank_zero_only, rank_zero_warn
 from pytorch_lightning.utilities.parsing import AttributeDict, flatten_dict, is_picklable
+from pytorch_lightning.utilities.xla_device_utils import XLADeviceUtils
 
 
 def _module_available(module_path: str) -> bool:
     """Testing if given module is avalaible in your env
 
-    >>> _module_available('system')
+    >>> _module_available('os')
     True
     >>> _module_available('bla.bla')
     False
     """
     mods = module_path.split('.')
     assert mods, 'nothing given to test'
     # it has to be tested as per partets
-    for i in range(1, len(mods)):
-        module_path = '.'.join(mods[:i])
+    for i in range(len(mods)):
+        module_path = '.'.join(mods[:i + 1])
         if importlib.util.find_spec(module_path) is None:
             return False
     return True
 
 
 APEX_AVAILABLE = _module_available("apex.amp")
-NATIVE_AMP_AVALAIBLE = hasattr(torch.cuda, "amp") and hasattr(torch.cuda.amp, "autocast")
+NATIVE_AMP_AVALAIBLE = _module_available("torch.cuda.amp")
+XLA_AVAILABLE = _module_available("torch_xla")
+HOROVOD_AVAILABLE = _module_available("horovod.torch")
+OMEGACONF_AVAILABLE = _module_available("omegaconf")
+HYDRA_AVAILABLE = _module_available("hydra")
```

Comment:
@tchaton @SeanNaren shall I make an introduction of each variable as separate PR?